test/e2e/space_test.go

```diff
@@ -0,0 +1,214 @@
+
+			err := memberAwait.WaitForMemberStatus(
+				wait.UntilMemberStatusHasConditions(ToolchainStatusComponentsNotReady("[routes]")))
+			require.NoError(t, err, "failed while waiting for MemberStatus to contain error due to che being required")
+
+			_, err = hostAwait.WaitForToolchainStatus(
+				wait.UntilToolchainStatusHasConditions(ToolchainStatusComponentsNotReady("[members]"), ToolchainStatusUnreadyNotificationNotCreated()))
+			require.NoError(t, err, "failed while waiting for ToolchainStatus to contain error due to che being required")
+
+			t.Run("verify member and toolchain status go back to ready", func(t *testing.T) {
+				// change che required flag back to true to resolve the error on the memberstatus
+				memberConfigurationWithCheRequired = testconfig.ModifyMemberOperatorConfigObj(memberAwait.GetMemberOperatorConfig(), testconfig.Che().Required(false))
+				hostAwait.UpdateToolchainConfig(testconfig.Members().Default(memberConfigurationWithCheRequired.Spec))
+
+				VerifyMemberStatus(t, memberAwait, consoleURL)
+				VerifyToolchainStatus(t, hostAwait, memberAwait)
+			})
+		})
+	})
+
+	memberAwait.WaitForUsersPodsWebhook()
+	memberAwait.WaitForAutoscalingBufferApp()
+	_, err := hostAwait.WaitForToolchainStatus(wait.UntilToolchainStatusHasConditions(
+		ToolchainStatusReadyAndUnreadyNotificationNotCreated()...))
+	require.NoError(t, err, "failed while waiting for ToolchainStatus")
+
+	rand.Seed(time.Now().UnixNano())
+
+	t.Run("create space", func(t *testing.T) {
+		// given
+		space := &toolchainv1alpha1.Space{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: hostAwait.Namespace,
+				Name:      "oddity-" + strconv.Itoa(rand.Int()),
+			},
+			Spec: toolchainv1alpha1.SpaceSpec{
+				TargetCluster: memberAwait.ClusterName,
+				TierName:      "base",
+			},
+		}
+
+		// when
+		err := hostAwait.Client.Create(context.TODO(), space)
+
+		// then
+		require.NoError(t, err)
+		// wait until NSTemplateSet has been created and Space is in `Ready` status
+		nsTmplSet, err := memberAwait.WaitForNSTmplSet(space.Name, wait.UntilNSTemplateSetHasConditions(Provisioned()))
+		require.NoError(t, err)
+		templateRefs := tiers.GetTemplateRefs(hostAwait, space.Spec.TierName)
+		for _, templateRef := range templateRefs.Namespaces {
+			_, err := memberAwait.WaitForNamespace(nsTmplSet.Name, templateRef, nsTmplSet.Spec.TierName)
+			require.NoError(t, err)
+		}
+		space, err = hostAwait.WaitForSpace(space.Name, wait.UntilSpaceHasConditions(Provisioned()))
+		require.NoError(t, err)
+
+		t.Run("delete space", func(t *testing.T) {
+			// now, delete the Space and expect that the NSTemplateSet will be deleted as well,
+			// along with its associated namespace
+
+			// when
+			err = hostAwait.Client.Delete(context.TODO(), space)
+
+			// then
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilSpaceDeleted(space.Name)
+			require.NoError(t, err)
+			err = memberAwait.WaitUntilNSTemplateSetDeleted(nsTmplSet.Name)
+			require.NoError(t, err)
```

Comment:
shouldn't we also check that the namespaces are deleted?

Reply:
added in cd9eb27